.github/workflows/swift-toolchain.yml

```diff
@@ -939,55 +950,100 @@ jobs:
-            $CACHE="Windows-aarch64.cmake"
-
-            # FIXME(compnerd) re-enable runtimes after we sort out compiler-rt
-            (Get-Content ${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-aarch64.cmake).Replace(' runtimes', '') | Set-Content ${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-aarch64.cmake
-          } else {
-            $CACHE="Windows-x86_64.cmake"
+          $CxxFlags = "${{ matrix.cxxflags }}"
+          $SwiftFlags = ""
+
+          if ( "${{ matrix.os }}" -eq "Windows" ) {
+            $SWIFTC = cygpath -m (Get-Command swiftc).Source
+            # Use toolchain clang to avoid broken __prefetch intrinsic on arm64 in Clang 18.
+            # TODO: Use llvm-19 when available. See https://github.com/compnerd/swift-build/issues/846
+            $CLANG_LOCATION = Split-Path (Get-Command swiftc).Source
+            $EXTRA_FLAGS = @(
+              "-D",
+              "SWIFT_NATIVE_SWIFT_TOOLS_PATH=${{ github.workspace }}/BinaryCache/0/bin"
+            )
+            if ( "${{ matrix.arch }}" -eq "arm64" ) {
+              $EXTRA_FLAGS += "${{ matrix.extra_flags }}".Split()
+              $EXTRA_FLAGS += @("-D", "CMAKE_SYSTEM_NAME=Windows")
+              $CACHE="${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-aarch64.cmake"
+
+              # FIXME(compnerd) re-enable runtimes after we sort out compiler-rt
+              (Get-Content ${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-aarch64.cmake).Replace(' runtimes', '') | Set-Content ${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-aarch64.cmake
+            } else {
+              # FIXME(steelskin) Setting `CMAKE_SYSTEM_PROCESSOR` and `CMAKE_SYSTEM_NAME` breaks the compiler-rt build.
+              $EXTRA_FLAGS += @("-D", "CMAKE_MT=mt")
+              $CACHE="${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-x86_64.cmake"
+            }
+            $CC = "cl"
+            $CXX = "cl"
+
+            # Workaround for compiler and STL version mismatch on Windows.
+            $CxxFlags += " -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH"
+            $SwiftFlags += " -Xcc -D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH"
+
+            $SDKROOT = cygpath -m ${env:SDKROOT}
+            $CLANG_LOCATION = cygpath -m $CLANG_LOCATION
+            $LIBPYTHON_PATH = "${env:PYTHON_LOCATION_${{ matrix.arch }}}/libs/python39.lib"
+            $PYTHON_INCLUDE_DIR = "${env:PYTHON_LOCATION_${{ matrix.arch }}}/include"
+            $PYTHON_BINARY="python.exe"
+            $ExeSuffix = ".exe"
+            Remove-Item env:\SDKROOT
+          } elseif ( "${{ matrix.os }}" -eq "Darwin" ) {
+            # Default swiftc comes from /usr/bin and is not compatible with the toolchain.
+            $CLANG_LOCATION = "${env:HOME}/Library/Developer/Toolchains/swift-6.0.1-RELEASE.xctoolchain/usr/bin"
+            $SWIFTC = Join-Path $CLANG_LOCATION "swiftc"
```

Comment:
This is the output from this command on the GHA bot:
```
2024-11-19 22:08:33.786 xcodebuild[5734:23049] [MT] DVTToolchain: Failed to load toolchain: <DVTFilePath:0x6000000643f0:'/Users/runner/Library/Developer/Toolchains/swift-6.0.1-RELEASE.xctoolchain'>: Error Domain=DVTToolchainErrorDomain Code=6 "The toolchain's Info.plist must contain a dictionary" UserInfo={NSLocalizedDescription=The toolchain's Info.plist must contain a dictionary}
2024-11-19 22:08:34.452 xcodebuild[5743:23090] [MT] DVTToolchain: Failed to load toolchain: <DVTFilePath:0x600002864630:'/Users/runner/Library/Developer/Toolchains/swift-6.0.1-RELEASE.xctoolchain'>: Error Domain=DVTToolchainErrorDomain Code=6 "The toolchain's Info.plist must contain a dictionary" UserInfo={NSLocalizedDescription=The toolchain's Info.plist must contain a dictionary}
/Applications/Xcode_15.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc
```

So we end up with the wrong toolchain because the command does not seem to be able to parse the `Info.plist` file.

Reply:
I used the environment variable value to configure the folder.